typings/index.d.ts

```diff
@@ -448,6 +448,13 @@ export class BitField<S extends string, N extends number | bigint = number> {
 
 export class ButtonInteraction<Cached extends CacheType = CacheType> extends MessageComponentInteraction<Cached> {
   private constructor(client: Client, data: RawMessageButtonInteractionData);
+  public readonly component: CacheTypeReducer<
+    Cached,
+    MessageButton,
+    APIButtonComponent,
+    MessageButton | APIButtonComponent,
```

Comment:
When can it be a raw component? We always build it, right?

Reply:
If an interaction is received from a guild the client is not in, it will be a "raw" interaction. The `message` property will be as-is without any parsing by discord.js, which means the `components` on it will also be raw:

https://github.com/discordjs/discord.js/blob/ee91f5a19cdcf15f17a80fc9a25a382b7cbcde9f/src/structures/MessageComponentInteraction.js#L23-L27